source/advanced-topics.adoc

```diff
@@ -408,6 +408,145 @@ an `(opts)` field. The `opts` (the string between parentheses) is passed
 exactly as is to getopt(3) for argc/argv processing at the beginning of
 a macro invocation.
 
+[[setup]]
+=== %setup
+
+With a bunch of source code tarballs used to build the package, the mighty
+``%setup`` macro and all its advanced features might come handy. But lets begin
+with its standard behavior. You can study it from the ``rpmbuild`` ouput. At the
+beginning of each phase, it outputs ``Executing(%something)``, e.g.:
+
+[source,bash]
+----
+Executing(%prep): /bin/sh -e /var/tmp/rpm-tmp.DhddsG
+----
+
+followed by output, similar to the output of shell when ``set -x`` is enabled,
+because it is actually shell output with ``set -x`` enabled. If that is not
+enough, it is possible to peek at content of ``/var/tmp/rpm-tmp.DhddsG``. For
+that, you have to run ``rpmbuild`` with ``--debug`` option, otherwise the file
```

Comment:
This sentence appears to consist of two independent clauses that should not be joined with a comma. Consider using a semicolon or period before otherwise instead.